### DISCUSSION TITLE
Generate the p0201_SYMPHONY.vbc file from Solver

### DISCUSSION THREAD

**Discussion by Bang2018** [coin-or/GrUMPy#19] [Q&A]
How to generate the p0201_SYMPHONY.vbc file from CBC/GLPK/Cplex/Gurobi solver? Please mention steps, if possible. 
I must say Ted it is an Excellent work.

**Comment by tkralphs** [comment]
(I Converted this to a Discussion rather than an Issue) 

VBC files have to be output by an instrumented solver during the solution process for a given instance. The idea is to record snapshots of the process so that the whole thing can later be visualized/simulated/analyzed. The particular file you referenced was output by a modified version of GLPK. The modification was done by someone else, who sent me the file, so I don't have access to that code. However, there are other solvers around that can output these files, such as [SYMPHONY](https://github.com/coin-or/SYMPHONY), which I maintain. If you want information about how to instrument an existing solver, I can say more about that, but you really just need to modify the solver to output the status lines that are required by the file format, which is documented. Let me know if I can be of further assistance. 

By the way, I have plans (which may or may not ever happen) to improve the way some of the visualizations are rendered, specifically to use a more modern graphics package. Right now, we are using gnuplot, which causes some issues. This stuff was implemented long ago when Python was less mature than now.

**Reply by Bang2018** [reply]
I am really happy to draw the attention. Yes, I am interested to use SYMPHONY instead of other solvers. Please let me know how to modify solver to get the output and status. I need your assistance. 
How to build the tree of Branch and Bound for real type business problem? Business problem contains multiple variables. 
I personally appreciate your plan to update visualizations. Use of Python packages are justified instead of gnuplot. I am willing to work under your supervisions. 
It is a wonderful work.
